lib/everypolitician.rb

```diff
@@ -116,6 +116,12 @@ def legislative_periods
     def directory
       @directory = raw_data[:sources_directory].split('/')[1,2].join('/')
     end
+
+    def self.from_sources_dir(dir)
+      @index_by_sources ||= EveryPolitician::countries.map(&:legislatures).flatten.group_by(&:directory)
+      # There must be a way to make this happen automatically in the line above
```

Comment:
It is. I put that comment in because pretty much every other time I've done something like that on two lines there's been a one line solution :)